scala3/scaladoc.md

```diff
@@ -54,16 +54,84 @@ For more information you can follow this [thread](https://contributors.scala-lan
+![](../resources/images/scala3/scaladoc/inkuire-1.0.0-M2_js_flatMap.gif)
 
+This feature is provided by [Inkuire](https://github.com/VirtusLab/Inkuire) search engine, which works for Scala 3 and Kotlin. To be up-to-date with development of this feature follow [Inkuire repository](https://github.com/VirtusLab/Inkuire) for new releases.
 
+Note that this feature is still in development, so it can be subject to considerable change.
+If You encounter a bug or have an idea for improvement don't hesitate to create an issue on [Inkuire](https://github.com/VirtusLab/Inkuire/issues/new) or [dotty](https://github.com/lampepfl/dotty/issues/new).
+
+
+Using this feature allows to find `def`s and `val`s from the documented code.
+
+In order for a scaladoc searchbar query to be searched using Inkuire instead of the default search engine, the query has to contain `=>` character sequence. 
+
+Accepted input is similar to a curried function signature in Scala 3. With some differences:
+- AndTypes, OrTypes and Functions have to be enclosed in parentheses e.g. `(Int & Any) => String`
+- Transformations without any input can be found by a syntactic by name call e.g. `=> Int`
+- A wildcard `_` can be used to indicate that we want to match any type in a given place e.g. `Long => Double => _`
+- Types in the form of single letter e.g. `A` or a letter with a digit `X1` are automatically assumed to be type variables
+- Other type variables can be declared just like in polymorphic functions e.g. `[A, B] => A => B => A` 
+
+Some examples to give You a better idea:
+- `List[Int] => (Int => Long) => List[Long]`
+- `(A, B) => A`
+- `Set[Long] => Long => Boolean`
+- `Int => Long => Int`
+- `String => Int => Char`
+- `(Int & Float) => (String | Double)`
+
+Inkuire works as a JavaScript worker in the browser thanks to the power of [ScalaJS](https://www.scala-js.org/).
+
+To enable Inkuire when running scaladoc add flag `-Ygenerate-inkuire`. By adding this flag two files are generated:
+- `inkuire-db.json` - this is the file containing all searchable declarations from the currently documented project in a format readable to Inkuire search engine.
+- `inkuire-config.json` - this file contains locations of database files that should be searchable from documentation of the current project. By default it will be generated with the location of the local db file as well as default implied locations of detabase files in `External mappings`.
+
+When it comes to how the code is mapped to InkuireDb entries, there are some transformations to make the engine more opinionated(though open to suggestions and changes). Firstly the receiver(non-module owner) of a function can be treated as a first argument. Automatic currying is also applied, so that the results aren't dependant on argument lists. When finding matches `val`s and `def`s are not distinguished.
```

Comment:
```suggestion
When it comes to how the code is mapped to InkuireDb entries, there are some transformations to make the engine more opinionated (though open to suggestions and changes). Firstly, the receiver (non-module owner) of a function can be treated as a first argument. Automatic currying is also applied, so that the results don’t depend on argument lists. When finding matches, `val`s and `def`s are not distinguished.
```

Reply:
done